fetcher-pipeline/app.py

```diff
@@ -74,17 +82,143 @@ def get_report(reports):
+                            # Now that we have file info for each, genearte final messages by merging common and unique properties into a single message of common + files of batch size N. Part of the for project_dir in language_dir loop. 500 is an arbitrary number that eems to given enough padding to not go over the 256 limit. 
+                            if common_message_data is not None:
+                                chunks = chunk_array(unique_message_data, 500)
+                                for chunk in chunks:
+                                    chunk_message = common_message_data.copy()
+                                    chunk_message["files"] = chunk
+                                    bus_messages.append(chunk_message)
+            logging.debug(f"Sending {len(bus_messages)} messages to queue")
+            if len(bus_messages) > 0:
+                # The whole loop isn't managed by asyncio, but we can run just this one coroutine with it. it'll set up and tear down an event loop as needed.  
+                asyncio.run(self.send_messages(bus_messages))
+                # self.send_messages(bus_messages)
+            else:
+                logging.debug("No messages to send")
+        except Exception as e:
+            logging.critical(f"Error sending messages to queue: {e.with_traceback()} {e}")
+            # blow up. If we can't send messages to bus, data not made availble in api. 
+            sys.exit(1)
+    async def send_messages(self, messages):
+        async with ServiceBusClient.from_connection_string(
+            conn_str=self.BUS_CONNECTION_STRING,
+            logging_enable=True
+        ) as service_bus_client:
+            sender = service_bus_client.get_topic_sender(topic_name=self.BUS_TOPIC)
+            async with sender:
+                # batch_message = await sender.create_message_batch()
+                # sender.
+                for message in messages:
+                    try:
+                        bus_message = ServiceBusMessage(
+                            json.dumps(message), 
+                            session_id=message["session_id"]
+                            );
+                        await sender.send_messages(bus_message)
+                        # batch_message.add_message(bus_message)
+                    except Exception as e:
+                        logging.error(e)
+                        logging.critical("Error sending messages to queue. Exiting since data won't make it to the api.")
+                        sys.exit(1)
+                        
+               
+                logging.debug(f"Done sending messages to queue. Sent {len(messages)} messages.")
+    
+    @staticmethod
+    def should_skip_file(file_path: Path, exlude_args: List[str]) -> bool:
+        should_skip = False
+        for arg in exlude_args:
+            if arg.startswith("."):
+                if file_path.suffix == arg:
+                    should_skip = True
```

Comment:
It would be more efficient to just `return True` here and in the other branches.  Then you can skip allocating `should_skip` and just `return False` at the end of the function.